src/bidiMapper/domains/network/NetworkStorage.ts

```diff
@@ -255,6 +259,23 @@ export class NetworkStorage {
     this.#blockedRequestMap.delete(requestId);
   }
 
+  /**
+   * Returns the blocked request associated with the given network ID, if any.
+   */
+  getBlockedRequest(networkId: Network.Request): {
+    request: Protocol.Fetch.RequestId;
+    phase?: Network.InterceptPhase; // TODO: make non-optional.
+    response?: Network.ResponseData; // TODO: make non-optional.
+  } {
+    const blockedRequest = this.#blockedRequestMap.get(networkId);
+    if (!blockedRequest) {
+      throw new NoSuchRequestException(
```

Comment:
this looks like it belongs to the NetworkProcessor. I suggest moving it there.

Reply:
Initially that was the case, but I moved it to storage.

Just moved it back.